### PR TITLE
Feat: Introduce HmctsrolesLibModule from @hmcts/hmctsroles-lib

### PR DESCRIPTION
**Pull request by masahmad**

### CODE REVIEW

File: src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -61,4 +61,15 @@ export class AuthService {
...
+
+    getLoggedInUserRoles(): string[] {
+
+        const jwt = this.cookieService.get(this.COOKIE_KEYS.TOKEN);
+        if (!jwt) {
+            return [];
+        }
+        const jwtData: any = this.decodeJwt(jwt);
+        const roles = jwtData.data;
+        return roles;
```
Comment (pardeepbasi) [inline src/app/auth/auth.service.ts]: probably can just return jwtData.data (very minor but no need to declare 'roles' if not needed)

File: src/app/shared/components/case-actions/case-actions.component.html [hunk 2]
```diff
@@ -4,6 +4,8 @@
             <h1 data-selector="heading" role="heading" aria-level="1" class="hmcts-page-heading__title govuk-heading-xl">{{header}}</h1>
         </div>
         <!-- TODO: Should use HMCTS Menu Component here, instead of this. -->
+        <div [unlessRole]="'caseworker-sscs-panelmember'"  roleList="{{roleList}}" roleSource="inline">
```
Comment (pardeepbasi) [inline src/app/shared/components/case-actions/case-actions.component.html]: just curious, do we always need to pass the roleList in? Perhaps there's another way the directive can obtain the roles (just means less maintenance for future code)

File: src/app/shared/components/case-actions/case-actions.component.spec.ts [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+import { Component, CUSTOM_ELEMENTS_SCHEMA, DebugElement, Input, ViewChild } from '@angular/core';
+import { FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { LinkItem } from '../../../domain/models/section_fields';
+import { Selector } from '../../selector-helper';
+
+// import { AuthService } from '../../../auth/auth.service';
+// import { AuthModule } from '../../../auth/auth.module';
+
+// class MockAuthService {
+//     getLoggedInUserRoles() {
```
Comment (frontr-uk) [inline src/app/shared/components/case-actions/case-actions.component.spec.ts]: can this be removed :)

File: src/app/shared/components/case-actions/case-actions.component.ts [hunk 4]
```diff
@@ -11,5 +12,11 @@ export class CaseActionsComponent {
     @Input() actionPrimaryButton: LinkItem;
     @Input() actionSecondaryButton: LinkItem;
     @Input() actionThirdButton: LinkItem;
-    constructor() { }
+    @Input() roleList: string[];
+    // roleList: string[];
+    // constructor(public authService: AuthService) {
```
Comment (frontr-uk) [inline src/app/shared/components/case-actions/case-actions.component.ts]: can this be removed

File: src/app/shared/components/question/question.component.html [hunk 5]
```diff
@@ -37,7 +37,10 @@ <h2 attr.data-selector="round-{{round.question_round_number}}" class="govuk-head
             <a data-selector="create-draft-questions-link" class="govuk-button" [ngClass]="{'jui-button--secondary':round.questions.length > 0, 'govuk-!-margin-bottom-4': true} " routerLink="./new/{{round.question_round_number}}">Add questions</a>
         </div>
         <div *ngIf="round.questions.length > 0" class="jui-button-action">
-            <a data-selector="send-draft-questions-link" class="govuk-button" routerLink="./check/{{round.question_round_number}}">Send questions</a>
+             
+
+            <a [unlessRole]="'caseworker-sscs-panelmember'"  roleList="{{roleList}}" roleSource="inline"
```
Comment (frontr-uk) [inline src/app/shared/components/question/question.component.html]: Best to have these as const and then typed, to make sure roles are correctly passed in.

File: src/app/shared/components/question/question.component.ts [hunk 6]
```diff
@@ -1,21 +1,28 @@
...
 
-export class QuestionComponent implements AfterViewInit  {
+export class QuestionComponent implements AfterViewInit {
     @Input() rounds = [];
     draft: boolean;
-    ngAfterViewInit () {
+
+    roleList: string[];
+    constructor(public authService: AuthService) {
+        this.roleList = authService.getLoggedInUserRoles();
```
Comment (frontr-uk) [inline src/app/shared/components/question/question.component.ts]: type?

File: src/app/shared/components/question/question.component.ts [hunk 7]
```diff
@@ -1,21 +1,28 @@
...
+import { AuthService } from '../../../auth/auth.service';
 
 @Component({
     selector: 'app-question',
     templateUrl: './question.component.html'
 })
 
-export class QuestionComponent implements AfterViewInit  {
+export class QuestionComponent implements AfterViewInit {
     @Input() rounds = [];
```
Comment (frontr-uk) [inline src/app/shared/components/question/question.component.ts]: type?

File: src/app/shared/components/case-actions/case-actions.component.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
-import {Component, Input} from '@angular/core';
-import {LinkItem} from '../../../domain/models/section_fields';
+import { Component, Input, OnInit } from '@angular/core';
+import { LinkItem } from '../../../domain/models/section_fields';
+//import { AuthService } from '../../../auth/auth.service';
```
Comment (frontr-uk) [inline src/app/shared/components/case-actions/case-actions.component.ts]: also this import can be removed

**Pull request merged**